src/game.cpp

```diff
@@ -5915,6 +5915,8 @@ void game::monmove()
 
         m.creature_in_field( critter );
 
+        mattack::lay_egg( &critter );
```

Comment:
Don't call it here, you need to bind the attack to the monster as it is the case for all other attacks.